CelesteNet.Client/Components/CelesteNetChatComponent.cs

```diff
@@ -198,19 +264,59 @@ public CelesteNetChatComponent(CelesteNetClientContext context, Game game)
                 // Enter - send.
                 // Handled in Update.
 
-            } else if (c == (char) 8) {
+            } else if (c == (char) 8 && _CursorIndex > 0) {
                 // Backspace - trim.
-                if (Typing.Length > 0)
-                    Typing = Typing.Substring(0, Typing.Length - 1);
+                if (Typing.Length > 0) {
+                    int trim = 1;
+
+                    // extra CursorIndex check since at index=1 using trim=1 is fine
+                    if (_ControlHeld && _CursorIndex > 1) {
+                        // adjust Ctrl+Backspace for having a space right before cursor
+                        int _adjustedCursor = CursorIndex;
+                        if (Typing[_CursorIndex - 1] == ' ')
+                            _adjustedCursor--;
+                        int prevWord = Typing.LastIndexOf(" ", _adjustedCursor - 1);
+                        // if control is held and a space is found, trim from cursor back to space
+                        if (prevWord >= 0)
+                            trim = _adjustedCursor - prevWord;
+                        // otherwise trim whole input back from cursor as it is one word
+                        else
+                            trim = _adjustedCursor;
+                    }
+                    // remove <trim> amount of characters before cursor
+                    Typing = Typing.Remove(_CursorIndex - trim, trim);
+                    _CursorIndex -= trim;
+                }
+                _RepeatIndex = 0;
+                _Time = 0;
+            } else if (c == (char) 127 && CursorIndex < Typing.Length) {
+                // Delete - remove character after cursor.
+                if (_ControlHeld && Typing[_CursorIndex] != ' ') {
+                    int nextWord = Typing.IndexOf(" ", _CursorIndex);
+                    // if control is held and a space is found, remove from cursor to space
+                    if (nextWord >= 0) {
+                        // include the found space in removal
+                        nextWord++;
+                        Typing = Typing.Remove(_CursorIndex, nextWord - _CursorIndex);
+                    } else {
+                        // otherwise remove everything after cursor
+                        Typing = Typing.Substring(0, _CursorIndex);
+                    }
+                } else {
+                    // just remove single char
+                    Typing = Typing.Remove(_CursorIndex, 1);
+                }
                 _RepeatIndex = 0;
                 _Time = 0;
-
-            } else if (c == (char) 127) {
-                // Delete - currenly not handled.
-
             } else if (!char.IsControl(c)) {
-                // Any other character - append.
-                Typing += c;
+                if(CursorIndex == Typing.Length) {
```

Comment:
Please change the spacing here to match the rest of the project 😅 

```suggestion
                if (CursorIndex == Typing.Length) {
```